engine/runtime/src/main/java/org/enso/interpreter/node/callable/dispatch/InvokeFunctionNode.java

```diff
@@ -224,4 +249,54 @@ public void setId(UUID id) {
   public UUID getId() {
     return functionCallInstrumentationNode.getId();
   }
+
+  /** Returns true if the given function is in the same project as this node. */
+  private boolean isInSameProject(Function function) {
+    var thisProjName = getThisProjectName();
+    var funcProjName = getFunctionProjectName(function);
+    if (thisProjName != null && funcProjName != null) {
+      return thisProjName.equals(funcProjName);
```

Comment:
Can this be used without `@TruffleBoundary`?

Shouldn't we rather use some `==` check? Have a `ModuleScope` associated with a `Package` and check the `==`? E.g. use `modScope.getModule().getPackage()` not its name?

Reply:
> Can this be used without `@TruffleBoundary`

No. I have reimplemented this check to check for `==` on `Package<TruffleFile>` in 47da50e553d38ddc356d3143ff0129bd066ad7bd